sources/train.py

```diff
@@ -150,27 +154,39 @@
                             "'medium' values"))
         sys.exit(1)
 
-    # Instance name (name + image size + network size)
-    instance_name = args.name + "_" + str(args.image_size) + "_" + args.network_size
+    if not args.model in ["feature_detection", "semantic_segmentation"]:
+        utils.logger.error("Unsupported model. Please consider ")
+        utils.logger.utils(("'feature_detection' or 'semantic_segmentation'"))
+        sys.exit(1)
+
     # Data path and repository management
     dataset_repo = os.path.join(args.datapath, args.dataset)
-    suffix = "" if not args.aggregate_label else "aggregate_"
-    training_name = "training_" + suffix + str(args.image_size)
-    validation_name = "validation_" + suffix + str(args.image_size)
-    os.makedirs(os.path.join(dataset_repo, training_name, 'images'), exist_ok=True)
-    os.makedirs(os.path.join(dataset_repo, training_name, 'labels'), exist_ok=True)
-    os.makedirs(os.path.join(dataset_repo, validation_name, 'images'), exist_ok=True)
-    os.makedirs(os.path.join(dataset_repo, validation_name, 'labels'), exist_ok=True)
-    training_filename = os.path.join(dataset_repo, training_name + '.json')
-    validation_filename = os.path.join(dataset_repo, validation_name + '.json')
+    input_repo = os.path.join(dataset_repo, "input")
```

Comment:
Maybe you can put this content into a dedicated function e.g. `prepare_folders` or something else. What do you think?

Reply:
Why not. It will clarify the `train.py` and `test.py` modules. These tedious code lines seems unavoidable though (let's hide them into `utils.py` :) ).